packages/angular/projects/ui-angular/src/lib/components/authenticator/components/authenticator/authenticator.component.ts

```diff
@@ -171,10 +171,25 @@ export class AuthenticatorComponent
   }
 
   public hasTabs() {
-    const route = this.authenticator.route;
+    const { route } = this.authenticator;
     return route === 'signIn' || route === 'signUp';
   }
 
+  public hasRouteComponent() {
```

Comment:
since we have a "pure function" check, I guess I need to double check here, do we want to make it a pure function?

Reply:
Talked offline; This is a pure function but a not a side effect, so this is good to go.

Since Angular is OOP based, this function cannot be made pure because output is affected by `this.*`. 

I took a quick googling, and [some people see](https://softwareengineering.stackexchange.com/questions/96429/getters-and-setters-in-functional-languages#comment174337_96429) `this` as an implicit input to the function, which will qualify the function as pure. Couldn't find more source though 🤷

Reply:
But nevertheless, tldr; this is idiomatic and good to go 🙏